### PR TITLE
fix(#9914): Add upload file max size limit

### PR DESCRIPTION
**Pull request by pengyejun**
### SUMMARY
CSV Upload - add max filesize limit.
To slove https://github.com/apache/incubator-superset/issues/9914

### TEST PLAN
CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/core_tests.py [hunk 1]
```diff
@@ -981,6 +981,43 @@ def test_import_csv(self):
...
+            fail_msg = f'Unable to upload CSV file "{f1}" to table "{table_name}"'
+            fail_msg2 = (
+                f"Error message: Exceeded the upload maximum size of {max_bytes} bytes"
+            )
+            content.append(data)
+            self.create_sample_csvfile(f1, content)
+            resp = self.upload_csv(f1, table_name)
+            self.assertIn(fail_msg, resp)
+            self.assertIn(fail_msg2, resp)
+            # make sure that john and empty string are replaced with None
```
Comment (bkyryliuk) [inline tests/core_tests.py:1013]: doesn't seem relevant

File: tests/core_tests.py [hunk 2]
```diff
@@ -981,6 +981,43 @@ def test_import_csv(self):
...
+        try:
+            success_msg_f1 = f'CSV file "{f1}" uploaded to table "{table_name}"'
+
+            # initial upload with fail mode
+            resp = self.upload_csv(f1, table_name)
+            self.assertIn(success_msg_f1, resp)
+
+            # upload again with fail mode; should fail
+            fail_msg = f'Unable to upload CSV file "{f1}" to table "{table_name}"'
+            fail_msg2 = (
```
Comment (bkyryliuk) [inline tests/core_tests.py:1005]: I think I am missing smth here, could you please add a comment explaining why 2nd upload exceeds the max bytes?

**Review by bkyryliuk — COMMENTED** [review]
overall looks good - just a small question about the unit test

File: tests/core_tests.py [hunk 3]
```diff
@@ -981,6 +981,43 @@ def test_import_csv(self):
...
+        if not max_bytes:
+            return
+        self.login(username="admin")
+        table_name = "".join(random.choice(string.ascii_lowercase) for _ in range(5))
+        f1 = "testSize.csv"
+        self.enable_csv_upload(utils.get_example_database())
+        repeat = max_bytes // 4
+        data = "a,b"
+        content = [data for _ in range(repeat)]
+        self.create_sample_csvfile(f1, content)
```
Comment (pengyejun) [inline tests/core_tests.py:995]: The single line content of the file is a,b\n.The size is 4byte.
repeat is the number of file lines.
file f1 less than or equal to UPLOAD_MAX_BYTES

File: tests/core_tests.py [hunk 4]
```diff
@@ -981,6 +981,43 @@ def test_import_csv(self):
...
+        try:
+            success_msg_f1 = f'CSV file "{f1}" uploaded to table "{table_name}"'
+
+            # initial upload with fail mode
+            resp = self.upload_csv(f1, table_name)
+            self.assertIn(success_msg_f1, resp)
```
Comment (pengyejun) [inline tests/core_tests.py:1001]: So the upload was successful

File: tests/core_tests.py (hunk 2 above)
Comment (pengyejun) [reply tests/core_tests.py:1005]: beacause content.append(data),file f1 is larger than UPLOAD_MAX_BYTES.
So the upload was fail

File: tests/core_tests.py (hunk 1 above)
Comment (pengyejun) [reply tests/core_tests.py:1013]: repeat-1 should be equal to the number of lines in the f1 csv file (excluding headers).
The number of rows in table should be equal to repeat-1.
The data form row content should be equal to ("a", "b")

**Comment by pengyejun** [comment]
> overall looks good - just a small question about the unit test

@bkyryliuk thank you for your reply. Please see my comment .But my English is not good. i am so sorry.

File: tests/core_tests.py (hunk 1 above)
Comment (bkyryliuk) [reply tests/core_tests.py:1032]: I meant the comment :)

**Comment by bkyryliuk** [comment]
@pengyejun thanks, looks mostly good, let's fix the formatting and would be good to go

**Comment by GerbenvdHuizen** [comment]
Would be nice to either reopen this PR, or have some other solution to limit the size of uploaded files for each form view.

**Pull request closed without merging**